In [1]:
import pandas as pd
import numpy as np

In [2]:
path = "/Users/juyeon/추천시스템_프로젝트/data/"

musinsa = pd.read_csv(path+"musinsa.csv")
print(musinsa.shape)
musinsa.head()

(65551, 26)


/opt/anaconda3/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3444: DtypeWarning: Columns (25) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


Column1          name main_category sub_category           brand  \
0        0   NM2DN51A_빅샷            가방           백팩  THE NORTH FACE   
1        1    THINK PACK            가방           백팩            YALE   
2        2  메쉬 스트링 백팩 블랙            가방           백팩       NEIKIDNIS   
3        3    앱솔루트 백팩 블랙            가방           백팩       NEIKIDNIS   
4        4  슈퍼브레이크 BLACK            가방           백팩        JANSPORT   

         number                                               tags     price  \
0      NM2DN51A           ['가방', '데이팩', '백팩', '빅샷', '신학기백팩', '캠핑']  150,000원   
1  YB1BG0031BK0  ['백팩', '노트북백팩', '백팩추천', '데일리백', '코듀라', '로고', '...  119,000원   
2    NBP005-101  ['메쉬스트링백팩', '스트링백팩', 'MESHSTRINGBACKPACK', '코듀...   69,000원   
3      037ASB06  ['백팩', '앱솔루트', '학생가방', '코듀라', '포켓', '노트북', '커플...   94,000원   
4   JS0A4QUT008  ['슈퍼브레이크', '백팩', '가방', '데일리', '여행', '아웃도어', '포켓']   49,000원   

   rating  rating_num  ... age18 age19_23  age24_28 age29_33 age34_39 age40  \
0     4.9     15173.0  ...  0.32     0.27      0.21     0.10     0.03  0.07   
1     4.9      3184.0  ...  0.41     0.27      0.16     0.05     0.02  0.09   
2     4.9      3733.0  ...  0.42     0.25      0.16     0.05     0.02  0.10   
3     4.8     12697.0  ...  0.44     0.18      0.16     0.08     0.03  0.11   
4     4.8      3823.0  ...  0.41     0.28      0.15     0.06     0.03  0.07   

    man  woman                                                img   _1  
0  0.60   0.40  https://image.msscdn.net/images/goods_img/2019...  NaN  
1  0.43   0.57  https://image.msscdn.net/images/goods_img/2021...  NaN  
2  0.44   0.56  https://image.msscdn.net/images/goods_img/2020...  NaN  
3  0.48   0.52  https://image.msscdn.net/images/goods_img/2017...  NaN  
4  0.23   0.77  https://image.msscdn.net/images/goods_img/2020...  NaN  

[5 rows x 26 columns]

In [3]:
musinsa = musinsa.iloc[:,1:]

In [4]:
musinsa

name main_category sub_category  \
0                               NM2DN51A_빅샷            가방           백팩   
1                                THINK PACK            가방           백팩   
2                              메쉬 스트링 백팩 블랙            가방           백팩   
3                                앱솔루트 백팩 블랙            가방           백팩   
4                              슈퍼브레이크 BLACK            가방           백팩   
...                                     ...           ...          ...   
65546  Comfortable Check Jogger Pants_Beige            바지   트레이닝/조거 팬츠   
65547                     헤비웨이트 스웨트 팬츠 [크림]            바지   트레이닝/조거 팬츠   
65548                        카고 조거 팬츠 (올리브)            바지   트레이닝/조거 팬츠   
65549           마일 기모 벌룬 스웨트팬츠 그레이 DELP4444            바지   트레이닝/조거 팬츠   
65550                     사이드 지퍼 스Ÿ‡팬츠 (블랙)            바지   트레이닝/조거 팬츠   

                  brand        number  \
0        THE NORTH FACE      NM2DN51A   
1                  YALE  YB1BG0031BK0   
2             NEIKIDNIS    NBP005-101   
3             NEIKIDNIS      037ASB06   
4              JANSPORT   JS0A4QUT008   
...                 ...           ...   
65546           ELLIOTI    EPT003A4BE   
65547  MUSINSA STANDARD   MMJPK611-CR   
65548        URBANSTOFF   U203-B501OV   
65549           PERSTEP       2161826   
65550             CLACO    CLACO-PT12   

                                                    tags     price  rating  \
0               ['가방', '데이팩', '백팩', '빅샷', '신학기백팩', '캠핑']  150,000원     4.9   
1      ['백팩', '노트북백팩', '백팩추천', '데일리백', '코듀라', '로고', '...  119,000원     4.9   
2      ['메쉬스트링백팩', '스트링백팩', 'MESHSTRINGBACKPACK', '코듀...   69,000원     4.9   
3      ['백팩', '앱솔루트', '학생가방', '코듀라', '포켓', '노트북', '커플...   94,000원     4.8   
4      ['슈퍼브레이크', '백팩', '가방', '데일리', '여행', '아웃도어', '포켓']   49,000원     4.8   
...                                                  ...       ...     ...   
65546          ['밴딩팬츠', '체크팬츠', '일자팬츠', '와이드팬츠', '체크바지']   45,000원     4.8   
65547  ['스웨트컬렉션', '이지웨어', '트레이닝복', '운동복', '원마일웨어', '스...   33,900원     4.8   
65548  ['카고팬츠', '조거팬츠', '카고조거팬츠', '코튼팬츠', '밴딩팬츠', '와이...   68,000원     4.7   
65549      ['팬츠', '스웨트팬츠', '긴바지', '바지', '스Ÿ‡팬츠', '밴딩팬츠']   39,800원     4.8   
65550  ['트랙팬츠', '조거팬츠', '트레이닝바지', '스웨트팬츠', '스Ÿ‡팬츠', '...   89,000원     4.8   

       rating_num    season  ... age18  age19_23 age24_28 age29_33 age34_39  \
0         15173.0  2021 ALL  ...  0.32      0.27     0.21     0.10     0.03   
1          3184.0  2022 ALL  ...  0.41      0.27     0.16     0.05     0.02   
2          3733.0       NaN  ...  0.42      0.25     0.16     0.05     0.02   
3         12697.0       NaN  ...  0.44      0.18     0.16     0.08     0.03   
4          3823.0       NaN  ...  0.41      0.28     0.15     0.06     0.03   
...           ...       ...  ...   ...       ...      ...      ...      ...   
65546       147.0  2021 F/W  ...  0.24      0.23     0.19     0.18     0.07   
65547      8807.0  2021 S/S  ...  0.15      0.25     0.22     0.16     0.10   
65548       177.0  2020 F/W  ...  0.13      0.18     0.23     0.18     0.12   
65549       205.0  2021 F/W  ...  0.21      0.31     0.29     0.10     0.03   
65550       130.0  2021 F/W  ...  0.21      0.17     0.29     0.18     0.06   

       age40   man  woman                                                img  \
0       0.07  0.60   0.40  https://image.msscdn.net/images/goods_img/2019...   
1       0.09  0.43   0.57  https://image.msscdn.net/images/goods_img/2021...   
2       0.10  0.44   0.56  https://image.msscdn.net/images/goods_img/2020...   
3       0.11  0.48   0.52  https://image.msscdn.net/images/goods_img/2017...   
4       0.07  0.23   0.77  https://image.msscdn.net/images/goods_img/2020...   
...      ...   ...    ...                                                ...   
65546   0.09  0.67   0.33  https://image.msscdn.net/images/goods_img/2021...   
65547   0.12  0.71   0.29  https://image.msscdn.net/images/goods_img/2021...   
65548   0.16  0.85   0.15  https://image.msscdn.net

## 전처리
- 52543 확인해보면 뒤로 한칸씩 밀렸음을 확인 할 수 있다. 이거 먼저 수정하고 전처리 진행한다.  

In [5]:
musinsa.loc[52543,]

name                            (VLADIMIR) 워터프루프 유틸리티 테크 셋업 (윈드자켓)
main_category                                                  아우터
sub_category                                                아노락 재킷
brand                                                DIMITRI BLACK
number                                                V20FAJP001BK
tags                                                  V20SPPL001BK
price            ['바람막이', '윈드브레이커', '조거팬츠', '카고조거팬츠', '트레이닝셋업',...
rating                                                         NaN
rating_num                                                     5.0
season                                                         131
gender                                                    2022 S/S
like                                                           NaN
view                                                           919
sale                                                     5.2천 회 이상
coordi                                                     200

In [6]:
price_1 = musinsa.loc[52543,]["price":]
price = list(price_1)
musinsa.loc[52543, "tags" : "img"] = price
musinsa.loc[52543, "_1"] = np.NaN
musinsa.loc[52543, "woman"]  = 0.61

In [7]:
musinsa.loc[52543,]

name                            (VLADIMIR) 워터프루프 유틸리티 테크 셋업 (윈드자켓)
main_category                                                  아우터
sub_category                                                아노락 재킷
brand                                                DIMITRI BLACK
number                                                V20FAJP001BK
tags             ['바람막이', '윈드브레이커', '조거팬츠', '카고조거팬츠', '트레이닝셋업',...
price                                                          NaN
rating                                                         5.0
rating_num                                                     131
season                                                    2022 S/S
gender                                                         NaN
like                                                           919
view                                                     5.2천 회 이상
sale                                                       200개 이상
coordi                                                        

In [8]:
new_musinsa = musinsa.astype({"price":"string", "rating_num":"float", "gender":"string", "like":"float", "coordi": "string"})

In [9]:
new_musinsa.to_csv(path + "new_musinsa.csv", index=True)

In [3]:
path = "/Users/juyeon/추천시스템_프로젝트/data/"

new_musinsa = pd.read_csv(path+"new_musinsa.csv")
new_musinsa = new_musinsa.iloc[:,1:]
print(new_musinsa.shape)
new_musinsa.head()

(65551, 25)


name main_category sub_category           brand        number  \
0   NM2DN51A_빅샷            가방           백팩  THE NORTH FACE      NM2DN51A   
1    THINK PACK            가방           백팩            YALE  YB1BG0031BK0   
2  메쉬 스트링 백팩 블랙            가방           백팩       NEIKIDNIS    NBP005-101   
3    앱솔루트 백팩 블랙            가방           백팩       NEIKIDNIS      037ASB06   
4  슈퍼브레이크 BLACK            가방           백팩        JANSPORT   JS0A4QUT008   

                                                tags     price  rating  \
0           ['가방', '데이팩', '백팩', '빅샷', '신학기백팩', '캠핑']  150,000원     4.9   
1  ['백팩', '노트북백팩', '백팩추천', '데일리백', '코듀라', '로고', '...  119,000원     4.9   
2  ['메쉬스트링백팩', '스트링백팩', 'MESHSTRINGBACKPACK', '코듀...   69,000원     4.9   
3  ['백팩', '앱솔루트', '학생가방', '코듀라', '포켓', '노트북', '커플...   94,000원     4.8   
4  ['슈퍼브레이크', '백팩', '가방', '데일리', '여행', '아웃도어', '포켓']   49,000원     4.8   

   rating_num    season  ... age18  age19_23 age24_28 age29_33 age34_39  \
0     15173.0  2021 ALL  ...  0.32      0.27     0.21     0.10     0.03   
1      3184.0  2022 ALL  ...  0.41      0.27     0.16     0.05     0.02   
2      3733.0       NaN  ...  0.42      0.25     0.16     0.05     0.02   
3     12697.0       NaN  ...  0.44      0.18     0.16     0.08     0.03   
4      3823.0       NaN  ...  0.41      0.28     0.15     0.06     0.03   

   age40   man  woman                                                img  _1  
0   0.07  0.60   0.40  https://image.msscdn.net/images/goods_img/2019... NaN  
1   0.09  0.43   0.57  https://image.msscdn.net/images/goods_img/2021... NaN  
2   0.10  0.44   0.56  https://image.msscdn.net/images/goods_img/2020... NaN  
3   0.11  0.48   0.52  https://image.msscdn.net/images/goods_img/2017... NaN  
4   0.07  0.23   0.77  https://image.msscdn.net/images/goods_img/2020... NaN  

[5 rows x 25 columns]

## 1. 코디 결측치 채우기

> - 1차 브랜드로 그룹핑 
> - 2차 인덱스 main_category + sub_category
> - 코디 forwardfiill 같은 걸로 결측값 채우기

- 이렇게 해결 안될 것 같음
> coordi 값 있는 것들 dictinary 형태로 만들고 
없는 값 채우는 형식으로 해보자 


In [4]:
#corrdi_값 있는 것들 따로 저장

coordi_notna = new_musinsa[new_musinsa["coordi"].notnull()][["brand", "main_category","sub_category","coordi"]].reset_index()

In [5]:
#key 값: brand+main_category+sub_category / value 값: coordi로 dictionary 생성
#brand, category 같아도 코디 다른 경우 있음 -> 어떤 것으로 대체? 일단 합칩합으로 실행  

import ast

coordi_notna_dic = {}
for i in range(coordi_notna.shape[0]):
    brand = coordi_notna.loc[i,"brand"]
    main = coordi_notna.loc[i,"main_category"]
    sub = coordi_notna.loc[i,"sub_category"]
    key = brand + main + sub
    value = ast.literal_eval(coordi_notna.loc[i,"coordi"])
    temp = coordi_notna_dic.get(key)
    if temp == None:
        coordi_notna_dic[key] = value
    else:
        coordi_notna_dic[key] = list(set(temp) | set(value))


In [6]:
coordi_notna_dic

{'THE NORTH FACE가방백팩': ['스포츠', '캐주얼', '스트릿'],
 'YALE가방백팩': ['캐주얼'],
 'NEIKIDNIS가방백팩': ['스포츠', '캐주얼', '스트릿'],
 'JANSPORT가방백팩': ['캐주얼'],
 'COVERNAT가방백팩': ['캐주얼'],
 'BUBILIAN가방백팩': ['스포츠', '캐주얼', '스트릿'],
 'MAISON MINED가방백팩': ['캐주얼'],
 'UMBRO가방백팩': ['스포츠', '캐주얼'],
 'MUSINSA STANDARD가방백팩': ['캐주얼'],
 'SHOOPEN가방백팩': ['캐주얼'],
 'KIRSH여성 가방백팩': ['캐주얼', '스트릿', '걸리시'],
 'ANDERSSON BELL가방백팩': ['캐주얼', '아메리칸 캐주얼', '스트릿'],
 'LEE가방백팩': ['캐주얼'],
 'THIRDWEAVE가방백팩': ['스포츠', '캐주얼', '스트릿'],
 'CARHARTT가방백팩': ['캐주얼'],
 'RAWROW가방백팩': ['캐주얼'],
 'ULTIMEIK가방백팩': ['스포츠', '캐주얼'],
 '4OUR B여성 가방백팩': ['캐주얼'],
 'BEN DAVIS가방백팩': ['캐주얼'],
 'ROIDESROIS가방백팩': ['캐주얼', '스트릿'],
 'JILL BY JILLSTUART가방백팩': ['캐주얼'],
 'MONSTER REPUBLIC가방백팩': ['캐주얼'],
 'SSERPE가방백팩': ['스트릿'],
 'SSRL가방백팩': ['캐주얼'],
 'NATIONAL PUBLICITY가방백팩': ['댄디', '포멀'],
 'THISISNEVERTHAT가방백팩': ['캐주얼', '스트릿'],
 'NATIONALGEOGRAPHIC가방백팩': ['스포츠', '캐주얼'],
 'DISCOVERY EXPEDITION가방백팩': ['스포츠'],
 'BEIPRIL여성 가방백팩': ['캐주얼', '걸리시'],
 'EASTPAK가방백팩': ['캐주얼'],
 'CARHARTT WIP여성

In [7]:
for i in range(new_musinsa.shape[0]):
    brand = new_musinsa.loc[i,"brand"]
    main = new_musinsa.loc[i,"main_category"]
    sub = new_musinsa.loc[i,"sub_category"]
    key = brand + main + sub
    if pd.isna(new_musinsa.loc[i, "coordi"]):
        try:
            new_musinsa.loc[i, "coordi"] = coordi_notna_dic[key]
        except:
            pass

In [8]:
new_musinsa["coordi"].isnull().sum()

35231

기존 개수보다 훨씬 줄어든 것을 알 수 있다. 

In [9]:
musinsa = new_musinsa

##2. main_category, sub_category 변경 및 스포츠잡화 삭제 
> 1. 스포츠잡화 삭제
> 2. main_category, sub_category 변경

In [10]:
#스포츠잡화 삭제 

index_잡화 = musinsa[musinsa["sub_category"] == "스포츠잡화"].index
musinsa.drop(index_잡화, inplace=True)
musinsa.reset_index(drop=True, inplace=True)
musinsa

name main_category sub_category  \
0                               NM2DN51A_빅샷            가방           백팩   
1                                THINK PACK            가방           백팩   
2                              메쉬 스트링 백팩 블랙            가방           백팩   
3                                앱솔루트 백팩 블랙            가방           백팩   
4                              슈퍼브레이크 BLACK            가방           백팩   
...                                     ...           ...          ...   
65462  Comfortable Check Jogger Pants_Beige            바지   트레이닝/조거 팬츠   
65463                     헤비웨이트 스웨트 팬츠 [크림]            바지   트레이닝/조거 팬츠   
65464                        카고 조거 팬츠 (올리브)            바지   트레이닝/조거 팬츠   
65465           마일 기모 벌룬 스웨트팬츠 그레이 DELP4444            바지   트레이닝/조거 팬츠   
65466                     사이드 지퍼 스Ÿ‡팬츠 (블랙)            바지   트레이닝/조거 팬츠   

                  brand        number  \
0        THE NORTH FACE      NM2DN51A   
1                  YALE  YB1BG0031BK0   
2             NEIKIDNIS    NBP005-101   
3             NEIKIDNIS      037ASB06   
4              JANSPORT   JS0A4QUT008   
...                 ...           ...   
65462           ELLIOTI    EPT003A4BE   
65463  MUSINSA STANDARD   MMJPK611-CR   
65464        URBANSTOFF   U203-B501OV   
65465           PERSTEP       2161826   
65466             CLACO    CLACO-PT12   

                                                    tags     price  rating  \
0               ['가방', '데이팩', '백팩', '빅샷', '신학기백팩', '캠핑']  150,000원     4.9   
1      ['백팩', '노트북백팩', '백팩추천', '데일리백', '코듀라', '로고', '...  119,000원     4.9   
2      ['메쉬스트링백팩', '스트링백팩', 'MESHSTRINGBACKPACK', '코듀...   69,000원     4.9   
3      ['백팩', '앱솔루트', '학생가방', '코듀라', '포켓', '노트북', '커플...   94,000원     4.8   
4      ['슈퍼브레이크', '백팩', '가방', '데일리', '여행', '아웃도어', '포켓']   49,000원     4.8   
...                                                  ...       ...     ...   
65462          ['밴딩팬츠', '체크팬츠', '일자팬츠', '와이드팬츠', '체크바지']   45,000원     4.8   
65463  ['스웨트컬렉션', '이지웨어', '트레이닝복', '운동복', '원마일웨어', '스...   33,900원     4.8   
65464  ['카고팬츠', '조거팬츠', '카고조거팬츠', '코튼팬츠', '밴딩팬츠', '와이...   68,000원     4.7   
65465      ['팬츠', '스웨트팬츠', '긴바지', '바지', '스Ÿ‡팬츠', '밴딩팬츠']   39,800원     4.8   
65466  ['트랙팬츠', '조거팬츠', '트레이닝바지', '스웨트팬츠', '스Ÿ‡팬츠', '...   89,000원     4.8   

       rating_num    season  ... age18  age19_23 age24_28 age29_33 age34_39  \
0         15173.0  2021 ALL  ...  0.32      0.27     0.21     0.10     0.03   
1          3184.0  2022 ALL  ...  0.41      0.27     0.16     0.05     0.02   
2          3733.0       NaN  ...  0.42      0.25     0.16     0.05     0.02   
3         12697.0       NaN  ...  0.44      0.18     0.16     0.08     0.03   
4          3823.0       NaN  ...  0.41      0.28     0.15     0.06     0.03   
...           ...       ...  ...   ...       ...      ...      ...      ...   
65462       147.0  2021 F/W  ...  0.24      0.23     0.19     0.18     0.07   
65463      8807.0  2021 S/S  ...  0.15      0.25     0.22     0.16     0.10   
65464       177.0  2020 F/W  ...  0.13      0.18     0.23     0.18     0.12   
65465       205.0  2021 F/W  ...  0.21      0.31     0.29     0.10     0.03   
65466       130.0  2021 F/W  ...  0.21      0.17     0.29     0.18     0.06   

       age40   man  woman                                                img  \
0       0.07  0.60   0.40  https://image.msscdn.net/images/goods_img/2019...   
1       0.09  0.43   0.57  https://image.msscdn.net/images/goods_img/2021...   
2       0.10  0.44   0.56  https://image.msscdn.net/images/goods_img/2020...   
3       0.11  0.48   0.52  https://image.msscdn.net/images/goods_img/2017...   
4       0.07  0.23   0.77  https://image.msscdn.net/images/goods_img/2020...   
...      ...   ...    ...                                                ...   
65462   0.09  0.67   0.33  https://image.msscdn.net/images/goods_img/2021...   
65463   0.12  0.71   0.29  https://image.msscdn.net/images/goods_img/2021...   
65464   0.16  0.85   0.15  https://image.msscdn.net

In [11]:
#main_category, sub_category 변경

musinsa.loc[musinsa[musinsa["main_category"] == '스포츠/용품'].index[:616],["main_category","sub_category"]] = ["스포츠가방","보스턴/드럼/더플백"]

In [12]:
musinsa.loc[musinsa[musinsa["main_category"] == '스포츠/용품'].index[:424],["main_category","sub_category"]] = ["스포츠모자", "선캡/바이저"]

In [13]:
musinsa.loc[musinsa[musinsa["main_category"] == '스포츠/용품'].index[:145],["main_category","sub_category"]] = ["스포츠신발", "워터스포츠화"]

나머지는 아우터로 확인해보니 스포츠/용품이 상위 카테고리고 하위 카테고리는 다른 상품의 카테고리와 동일 하지만 지금 카테고리가 아우터에서 가디건인지 아노락 재킷인지.. 등 구분하기가 어려움 이 부분은 결측값으로 바꿔준 다음 fillna를 통해 (method="ffill
") 앞에 있는 데이터 값과 동일하게 채워준다. 

In [14]:
musinsa.loc[musinsa[musinsa["main_category"] == '스포츠/용품'].index,["main_category","sub_category"]] = np.NaN

In [15]:
#결측값 위의 값으로 채워준다.
musinsa["main_category"].fillna(method="ffill", inplace=True)
musinsa["sub_category"].fillna(method="ffill", inplace=True)

In [16]:
musinsa["main_category"].unique() 

array(['가방', '여성 가방', '스포츠가방', '모자', '스포츠모자', '상의', '스니커즈', '스커트', '신발',
       '스포츠신발', '아우터', '원피스', '바지', '하의'], dtype=object)

- 가방과 여성가방 통일 
- 바지 + 하의 = 하의로 통일


In [17]:
musinsa.loc[musinsa[musinsa["main_category"] == '여성 가방'].index,"main_category"] = "가방"

musinsa.loc[musinsa[musinsa["main_category"] == '바지'].index,"main_category"] = "하의"

In [18]:
musinsa["main_category"].unique() 

array(['가방', '스포츠가방', '모자', '스포츠모자', '상의', '스니커즈', '스커트', '신발', '스포츠신발',
       '아우터', '원피스', '하의'], dtype=object)

In [19]:
musinsa["sub_category"].unique()

array(['백팩', '캐리어', '가방 소품', '지갑/머니클립', '클러치 백', '메신저/크로스 백', '크로스백',
       '숄더백', '토트백', '에코백', '보스턴/드럼/더플백', '웨이스트 백', '파우치 백', '브리프케이스',
       '캡/야구 모자', '선캡/바이저', '헌팅캡/베레모', '페도라', '버킷/사파리햇', '비니', '기타 모자',
       '트루퍼', '반소매 티셔츠', '긴소매 티셔츠', '민소매 티셔츠', '셔츠/블라우스', '맨투맨/스웨트셔츠',
       '후드 티셔츠', '니트/스웨터', '기타 상의', '피케/카라 티셔츠', '캔버스/단화', '패션스니커즈화',
       '기타 스니커즈', '롱스커트', '미니스커트', '미디스커트', '구두', '부츠', '신발 용품', '로퍼',
       '힐/펌프스', '플랫 슈즈', '블로퍼', '샌들', '슬리퍼', '기타 신발', '워터스포츠화',
       '모카신/보트 슈즈', '환절기 코트', '겨울 싱글 코트', '겨울 더블 코트', '겨울 기타 코트',
       '롱패딩/롱헤비 아우터', '숏패딩/숏헤비 아우터', '\x08패딩베스트', '베스트', '사파리/헌팅 재킷',
       '나일론/코치  재킷', '기타 아우터', '후드 집업', '블루종/MA-1', '레더/라이더스 재킷', '무스탕/퍼',
       '트러커 재킷', '슈트/블레이저 재킷', '카디건', '아노락 재킷', '플리스/뽀글이', '트레이닝 재킷',
       '스타디움 재킷', '맥시 원피스', '미니원피스', '미디원피스', '기타 바지', '데님 팬츠', '레깅스',
       '숏 팬츠', '슈트 팬츠/슬랙스', '점프 슈트/오버올', '코튼 팬츠', '트레이닝/조거 팬츠'],
      dtype=object)

In [20]:
#패딩베스트 값 이상 앞에 바꿔주기
import re
musinsa["sub_category"] = [re.sub(r'[\x08]','', s) for s in musinsa["sub_category"]]

## 3. 데이터 제거 
> 1. tags 값과 coordi 값 두개 다 없는 행 삭제 
> 2. rating num median 그 이하 버리기 

In [21]:
#1.

# tags, coordi 없는 값 개수 확인 
musinsa[(musinsa["tags"] == "[]") & (musinsa["coordi"].isnull())].shape

(7976, 25)

7976개 삭제 가능

In [22]:
# 삭제 
both_nan = musinsa[(musinsa["tags"] == "[]") & (musinsa["coordi"].isnull())].index
musinsa.drop(both_nan, inplace=True)
musinsa.reset_index(drop=True, inplace=True)
musinsa

name main_category sub_category  \
0                               NM2DN51A_빅샷            가방           백팩   
1                                THINK PACK            가방           백팩   
2                              메쉬 스트링 백팩 블랙            가방           백팩   
3                                앱솔루트 백팩 블랙            가방           백팩   
4                              슈퍼브레이크 BLACK            가방           백팩   
...                                     ...           ...          ...   
57486  Comfortable Check Jogger Pants_Beige            하의   트레이닝/조거 팬츠   
57487                     헤비웨이트 스웨트 팬츠 [크림]            하의   트레이닝/조거 팬츠   
57488                        카고 조거 팬츠 (올리브)            하의   트레이닝/조거 팬츠   
57489           마일 기모 벌룬 스웨트팬츠 그레이 DELP4444            하의   트레이닝/조거 팬츠   
57490                     사이드 지퍼 스Ÿ‡팬츠 (블랙)            하의   트레이닝/조거 팬츠   

                  brand        number  \
0        THE NORTH FACE      NM2DN51A   
1                  YALE  YB1BG0031BK0   
2             NEIKIDNIS    NBP005-101   
3             NEIKIDNIS      037ASB06   
4              JANSPORT   JS0A4QUT008   
...                 ...           ...   
57486           ELLIOTI    EPT003A4BE   
57487  MUSINSA STANDARD   MMJPK611-CR   
57488        URBANSTOFF   U203-B501OV   
57489           PERSTEP       2161826   
57490             CLACO    CLACO-PT12   

                                                    tags     price  rating  \
0               ['가방', '데이팩', '백팩', '빅샷', '신학기백팩', '캠핑']  150,000원     4.9   
1      ['백팩', '노트북백팩', '백팩추천', '데일리백', '코듀라', '로고', '...  119,000원     4.9   
2      ['메쉬스트링백팩', '스트링백팩', 'MESHSTRINGBACKPACK', '코듀...   69,000원     4.9   
3      ['백팩', '앱솔루트', '학생가방', '코듀라', '포켓', '노트북', '커플...   94,000원     4.8   
4      ['슈퍼브레이크', '백팩', '가방', '데일리', '여행', '아웃도어', '포켓']   49,000원     4.8   
...                                                  ...       ...     ...   
57486          ['밴딩팬츠', '체크팬츠', '일자팬츠', '와이드팬츠', '체크바지']   45,000원     4.8   
57487  ['스웨트컬렉션', '이지웨어', '트레이닝복', '운동복', '원마일웨어', '스...   33,900원     4.8   
57488  ['카고팬츠', '조거팬츠', '카고조거팬츠', '코튼팬츠', '밴딩팬츠', '와이...   68,000원     4.7   
57489      ['팬츠', '스웨트팬츠', '긴바지', '바지', '스Ÿ‡팬츠', '밴딩팬츠']   39,800원     4.8   
57490  ['트랙팬츠', '조거팬츠', '트레이닝바지', '스웨트팬츠', '스Ÿ‡팬츠', '...   89,000원     4.8   

       rating_num    season  ... age18  age19_23 age24_28 age29_33 age34_39  \
0         15173.0  2021 ALL  ...  0.32      0.27     0.21     0.10     0.03   
1          3184.0  2022 ALL  ...  0.41      0.27     0.16     0.05     0.02   
2          3733.0       NaN  ...  0.42      0.25     0.16     0.05     0.02   
3         12697.0       NaN  ...  0.44      0.18     0.16     0.08     0.03   
4          3823.0       NaN  ...  0.41      0.28     0.15     0.06     0.03   
...           ...       ...  ...   ...       ...      ...      ...      ...   
57486       147.0  2021 F/W  ...  0.24      0.23     0.19     0.18     0.07   
57487      8807.0  2021 S/S  ...  0.15      0.25     0.22     0.16     0.10   
57488       177.0  2020 F/W  ...  0.13      0.18     0.23     0.18     0.12   
57489       205.0  2021 F/W  ...  0.21      0.31     0.29     0.10     0.03   
57490       130.0  2021 F/W  ...  0.21      0.17     0.29     0.18     0.06   

       age40   man  woman                                                img  \
0       0.07  0.60   0.40  https://image.msscdn.net/images/goods_img/2019...   
1       0.09  0.43   0.57  https://image.msscdn.net/images/goods_img/2021...   
2       0.10  0.44   0.56  https://image.msscdn.net/images/goods_img/2020...   
3       0.11  0.48   0.52  https://image.msscdn.net/images/goods_img/2017...   
4       0.07  0.23   0.77  https://image.msscdn.net/images/goods_img/2020...   
...      ...   ...    ...                                                ...   
57486   0.09  0.67   0.33  https://image.msscdn.net/images/goods_img/2021...   
57487   0.12  0.71   0.29  https://image.msscdn.net/images/goods_img/2021...   
57488   0.16  0.85   0.15  https://image.msscdn.net

행 개수: 65467개에서 (65467 - 7976 = 57491) 57491개 남았다.  

In [23]:
#2. 

#rating_num 결측치 채우기 
musinsa["rating_num"].fillna(0, inplace=True)

In [24]:
musinsa["rating_num"].median()

21.0

In [25]:
musinsa_21 = musinsa[(musinsa['rating_num'] > 21)]

In [26]:
musinsa_21.shape

(28344, 25)

In [27]:
musinsa_21.reset_index(drop=True, inplace=True)

중앙값 21 이상인 것만 추출해오면 28344개 나오는 것을 알 수 있다. (musinsa_21)

## 4. season

> 처리 
> - 2016 ALL    
> - 2017 ALL   /   2017 S/S  
> - 2018 ALL   /   2018 S/S   /   2018 F/W   
> - 2019 ALL   /   2019 S/S   /   2019 F/W
> - 2020 ALL   /   2020 S/S   /  2020 F/W 
> - 2021 ALL   / 2021 S/S     /   2021 F/W
> - 2022 ALL(2022) /   2022 S/S   /  2022 F/W
> - ALL(ALL ALL)
> - ALL S/S
> - ALL F/W
> - nan, 병행수입은 따로,, 보자

In [28]:
musinsa_21["season"].unique()

array(['2021 ALL', '2022 ALL', nan, '2020 ALL', '2021 S/S', '2019 ALL',
       '병행수입', '2022 S/S', 'ALL ALL', '2022', '2017 ALL', '2018 ALL',
       '2021 F/W', '2016 ALL', '2020 S/S', 'ALL', '2020 F/W', 'ALL S/S',
       'ALL F/W', '2022 F/W', '2019 S/S', '2019 F/W', '2018 S/S',
       '2018 F/W', '2017 S/S'], dtype=object)

In [29]:
# 2022를 2022 ALL로
musinsa_21.loc[musinsa_21[musinsa_21["season"] == "2022"].index, "season"] = "2022 ALL"

# ALL ALL 을 ALL로
musinsa_21.loc[musinsa_21[musinsa_21["season"] == "ALL ALL"].index, "season"] = "ALL"

/opt/anaconda3/lib/python3.9/site-packages/pandas/core/indexing.py:1817: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(loc, value, pi)


In [30]:
musinsa_21["season"].unique()

array(['2021 ALL', '2022 ALL', nan, '2020 ALL', '2021 S/S', '2019 ALL',
       '병행수입', '2022 S/S', 'ALL', '2017 ALL', '2018 ALL', '2021 F/W',
       '2016 ALL', '2020 S/S', '2020 F/W', 'ALL S/S', 'ALL F/W',
       '2022 F/W', '2019 S/S', '2019 F/W', '2018 S/S', '2018 F/W',
       '2017 S/S'], dtype=object)

병행수입도 다시 대체해 줄 예정 결측값으로 바꾼후 같이 변환한다.

In [31]:
musinsa_21.loc[musinsa_21[musinsa_21["season"] == "병행수입"].index, "season"] = np.NaN

In [32]:
#nan값 처리

#1. 가방은 다 ALL로
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["main_category"] == "가방")].index, "season"] = "ALL"
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["main_category"] == "스포츠가방")].index, "season"] = "ALL"

In [33]:
#2. 모자도 다 ALL
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["main_category"] == "모자")].index, "season"] = "ALL"
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["main_category"] == "스포츠모자")].index, "season"] = "ALL"

In [34]:
#3. 신발도 다 ALL
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["main_category"] == "신발")].index, "season"] = "ALL"
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["main_category"] == "스포츠신발")].index, "season"] = "ALL"
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["main_category"] == "스니커즈")].index, "season"] = "ALL"

In [35]:
musinsa_21[(pd.isna(musinsa_21["season"]))]["sub_category"].unique()

array(['반소매 티셔츠', '긴소매 티셔츠', '민소매 티셔츠', '셔츠/블라우스', '맨투맨/스웨트셔츠', '후드 티셔츠',
       '니트/스웨터', '기타 상의', '피케/카라 티셔츠', '롱스커트', '미니스커트', '미디스커트', '환절기 코트',
       '겨울 싱글 코트', '겨울 더블 코트', '겨울 기타 코트', '롱패딩/롱헤비 아우터', '숏패딩/숏헤비 아우터',
       '패딩베스트', '베스트', '사파리/헌팅 재킷', '나일론/코치  재킷', '기타 아우터', '후드 집업',
       '블루종/MA-1', '레더/라이더스 재킷', '무스탕/퍼', '트러커 재킷', '슈트/블레이저 재킷', '카디건',
       '아노락 재킷', '플리스/뽀글이', '트레이닝 재킷', '스타디움 재킷', '맥시 원피스', '미니원피스',
       '미디원피스', '기타 바지', '데님 팬츠', '레깅스', '숏 팬츠', '슈트 팬츠/슬랙스', '점프 슈트/오버올',
       '코튼 팬츠', '트레이닝/조거 팬츠'], dtype=object)

결측값이 아닌 값들의 시즌들을 확인해 적합한 시즌 찾아서 넣어주기 

In [83]:
#상의
musinsa_21[musinsa_21["sub_category"] == "반소매 티셔츠"]["season"].value_counts() # ALL S/S
musinsa_21[musinsa_21["sub_category"] == "긴소매 티셔츠"]["season"].value_counts() # ALL
musinsa_21[musinsa_21["sub_category"] == "민소매 티셔츠"]["season"].value_counts() # ALL S/S
musinsa_21[musinsa_21["sub_category"] == "셔츠/블라우스"]["season"].value_counts() # ALL
musinsa_21[musinsa_21["sub_category"] == "맨투맨/스웨트셔츠"]["season"].value_counts() # ALL F/W
musinsa_21[musinsa_21["sub_category"] == "후드 티셔츠"]["season"].value_counts() # ALL F/W
musinsa_21[musinsa_21["sub_category"] == "니트/스웨터"]["season"].value_counts() # ALL
musinsa_21[musinsa_21["sub_category"] == "기타 상의"]["season"].value_counts() # ALL
musinsa_21[musinsa_21["sub_category"] == "피케/카라 티셔츠"]["season"].value_counts() # ALL S/S

#스커트
musinsa_21[musinsa_21["sub_category"] == "롱스커트"]["season"].value_counts() # ALL
musinsa_21[musinsa_21["sub_category"] == "미니스커트"]["season"].value_counts() # ALL
musinsa_21[musinsa_21["sub_category"] == "미디스커트"]["season"].value_counts() # ALL

#아우터
musinsa_21[musinsa_21["sub_category"] == "환절기 코트"]["season"].value_counts() # ALL
musinsa_21[musinsa_21["sub_category"] == "겨울 싱글 코트"]["season"].value_counts() # ALL F/W
musinsa_21[musinsa_21["sub_category"] == "겨울 더블 코트"]["season"].value_counts() # ALL F/W
musinsa_21[musinsa_21["sub_category"] == "겨울 기타 코트"]["season"].value_counts() # ALL F/W
musinsa_21[musinsa_21["sub_category"] == "롱패딩/롱헤비 아우터"]["season"].value_counts() # ALL F/W
musinsa_21[musinsa_21["sub_category"] == "숏패딩/숏헤비 아우터"]["season"].value_counts() # ALL F/W
musinsa_21[musinsa_21["sub_category"] == "패딩베스트"]["season"].value_counts() # ALL F/W
musinsa_21[musinsa_21["sub_category"] == "베스트"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "사파리/헌팅 재킷"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "나일론/코치  재킷"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "기타 아우터"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "후드 집업"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "블루종/MA-1"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "레더/라이더스 재킷"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "무스탕/퍼"]["season"].value_counts() # ALL F/W
musinsa_21[musinsa_21["sub_category"] == "트러커 재킷"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "슈트/블레이저 재킷"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "카디건"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "아노락 재킷"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "플리스/뽀글이"]["season"].value_counts() # ALL F/W
musinsa_21[musinsa_21["sub_category"] == "트레이닝 재킷"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "스타디움 재킷"]["season"].value_counts() # ALL 

#원피스
musinsa_21[musinsa_21["sub_category"] == "맥시 원피스"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "미니원피스"]["season"].value_counts() # ALL
musinsa_21[musinsa_21["sub_category"] == "미디원피스"]["season"].value_counts() # ALL

#하의
musinsa_21[musinsa_21["sub_category"] == "기타 바지"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "데님 팬츠"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "레깅스"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "숏 팬츠"]["season"].value_counts() # ALL S/S
musinsa_21[musinsa_21["sub_category"] == "슈트 팬츠/슬랙스"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "점프 슈트/오버올"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "코튼 팬츠"]["season"].value_counts() # ALL 
musinsa_21[musinsa_21["sub_category"] == "트레이닝/조거 팬츠"]["season"].value_counts() # ALL 

ALL         206
2022 S/S    203
2021 F/W    188
2021 S/S    173
2020 F/W     56
2020 S/S     28
ALL F/W       8
2022 ALL      7
ALL S/S       6
2019 F/W      5
2021 ALL      2
2022 F/W      2
Name: season, dtype: int64

In [36]:
#상의
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["sub_category"].isin(["긴소매 티셔츠","셔츠/블라우스","니트/스웨터","기타 상의"]))].index, "season"] = "ALL"
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["sub_category"].isin(["반소매 티셔츠","민소매 티셔츠","피케/카라 티셔츠"]))].index, "season"] = "ALL S/S"
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["sub_category"].isin(["맨투맨/스웨트셔츠","후드 티셔츠"]))].index, "season"] = "ALL F/W"

#스커트
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["main_category"] == "스커트")].index, "season"] = "ALL"

#아우터
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["sub_category"].isin(["환절기 코트","베스트","사파리/헌팅 재킷","나일론/코치  재킷","기타 아우터","후드 집업","블루종/MA-1","레더/라이더스 재킷","트러커 재킷","슈트/블레이저 재킷","카디건","아노락 재킷","트레이닝 재킷","스타디움 재킷"]))].index, "season"] = "ALL"
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["sub_category"].isin(["겨울 싱글 코트","겨울 더블 코트","겨울 기타 코트","롱패딩/롱헤비 아우터","숏패딩/숏헤비 아우터","패딩베스트","무스탕/퍼","플리스/뽀글이"]))].index, "season"] = "ALL F/W"

#원피스
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["main_category"] == "원피스")].index, "season"] = "ALL"

#하의
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["sub_category"].isin(["기타 바지","데님 팬츠","레깅스","슈트 팬츠/슬랙스","점프 슈트/오버올","코튼 팬츠","트레이닝/조거 팬츠"]))].index, "season"] = "ALL"
musinsa_21.loc[musinsa_21[(pd.isna(musinsa_21["season"])) & (musinsa_21["sub_category"].isin(["숏 팬츠"]))].index, "season"] = "ALL S/S"

In [37]:
musinsa_21["season"].isnull().sum()

0

## 데이터 더 삭제 
> - 좋아요수 중앙값 초과값 & 조회수 중앙값 초과값 만 데이터 가지고 오기 
> - 조회수는 4만 개이상 이런식으로 되어있음 => 전처리 후 중앙값 구하기(4만개 이상이면 -> 40000, 3천개 이상 -> 3000이렇게 전처리)

In [90]:
musinsa_21["like"].median()

617.0

In [104]:
musinsa_21[musinsa_21["like"] > 617]["main_category"].value_counts()

상의       4188
아우터      3067
하의       2816
가방       1430
스니커즈      894
신발        830
모자        604
스커트       158
원피스       154
스포츠가방      14
스포츠모자       5
스포츠신발       4
Name: main_category, dtype: int64

In [95]:
df = musinsa_21

df['view']=df['view'].str.replace(' 회 이상', '', regex=False)
df['view']=df['view'].str.replace(' 이상', '', regex=False)

for i, v in enumerate(df['view']):
    v = str(v)
    if v[-1]=='회':
        df.loc[i, 'view'] = float(v[:-1])
    elif v[-1] == "천":
        df.loc[i, 'view'] = float(v[:-1])*1000
    elif v[-1] == "만":
        df.loc[i, 'view'] = float(v[:-1])*10000
    elif v == np.nan:
        df.loc[i, 'view'] = -1

df.view = df.view.apply(float)

/var/folders/qm/gf24s1j90lb18c61ht3lq1jm0000gn/T/ipykernel_83543/1572057185.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['view']=df['view'].str.replace(' 회 이상', '', regex=False)
/var/folders/qm/gf24s1j90lb18c61ht3lq1jm0000gn/T/ipykernel_83543/1572057185.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['view']=df['view'].str.replace(' 이상', '', regex=False)
/opt/anaconda3/lib/python3.9/site-packages/pandas/core/indexing.py:1817: SettingWithCopyWarning: 
A value is trying to be set on a copy of

In [98]:
musinsa_21["view"].median()

1300.0

In [108]:
musinsa_21[(musinsa_21["view"] > 1300) & (musinsa_21["like"] > 617)]["main_category"].value_counts()

상의       2952
하의       2227
아우터      1641
가방        797
스니커즈      664
신발        597
모자        397
원피스        99
스커트        73
스포츠가방      10
스포츠모자       5
스포츠신발       4
Name: main_category, dtype: int64

In [117]:
musinsa_9466 = musinsa_21[(musinsa_21["view"] > 1300) & (musinsa_21["like"] > 617)]

In [118]:
musinsa_9466.reset_index(drop=True, inplace=True)

In [119]:
musinsa_9466.to_csv(path + "musinsa_9466.csv", index=True)

최종적으로 musinsa_6466 데이터 사용한다. -> 회의를 통해 결정된 데이터

##  이후 전처리에 이용된 데이터는 musinsa_6466이 아니라 musinsa_21로 실행했다. 

## 6. view - like - sale 의 관계 
- 좋아요 수가 판매로 이어졌는지 확인하는 피쳐 생성
- 조회수 대비 좋아요  (view => like)
- 좋아요 대비 조회수  (like => sale)
- 판매량 대비 리뷰수  (sale => rating_num)

#### 문제1. 리뷰 수가 있음에도 조회수가 결측값인 경우가 존재한다. 
> 리뷰 수 최대값이 19264개 조회수 결측값을 0으로 채우는 게 좋을까(대체적으로 적긴하다.)



In [38]:
musinsa_21[musinsa_21["view"].isnull()]["rating_num"].describe()

count      884.000000
mean       257.128959
std       1118.858992
min         22.000000
25%         31.750000
50%         50.000000
75%        109.000000
max      19264.000000
Name: rating_num, dtype: float64

## 7. tags, coordi 결측값 확인 & TF-IDF 실행해보기



In [39]:
#tags
import ast

a = []
num_na = 0

for i in musinsa_21["tags"]:
    li = ast.literal_eval(i)
    for j in li:
        a.append(j)
    if len(li) == 0:
        num_na += 1

In [40]:
print(len(a))
print(len(list(set(a))))
print(num_na)

165561
8860
1436


In [41]:
#coordi

a = []
num_na = 0

for i in musinsa_21["coordi"]:
    try:
        li = ast.literal_eval(i)
        for j in li:
            a.append(j)
    except:
        num_na += 1

In [42]:
print(len(a))
print(len(list(set(a))))
print(num_na)

22812
12
12236


###### tags는 너무 많으니까 보류 
###### coordi는 고유값 12개 TF-IDF 해보자 

In [43]:
#tags는 너무 많으니까 보류 
#coordi는 고유값 12개 TF-IDF 해보자 

In [44]:
total_count = len(musinsa_21.index)
total_coordi = list(set(a))

In [45]:
coordi_count = dict.fromkeys(total_coordi)

for each_coordi in musinsa_21["coordi"]:
    try:
        for coordi in ast.literal_eval(each_coordi):
            if coordi_count[coordi] == None:
                coordi_count[coordi] = 1
            else:
                coordi_count[coordi] += 1
    except:
        pass

In [46]:
coordi_count

{'레트로': 79,
 '홈웨어': 264,
 '아메리칸 캐주얼': 1276,
 '골프': 266,
 '댄디': 3009,
 '스포츠': 1420,
 '캐주얼': 10309,
 '포멀': 2616,
 '로맨틱': 514,
 '시크': 221,
 '스트릿': 1826,
 '걸리시': 1012}

In [47]:
## IDF 해당하는 부분 -> 장르별 가중치 생성하겠다. 

for coordi in coordi_count:
    coordi_count[coordi] = np.log10(total_count/coordi_count[coordi]) 
  
coordi_count 

{'레트로': 2.5548340480346794,
 '홈웨어': 2.03085721245529,
 '아메리칸 캐주얼': 1.3466104649399773,
 '골프': 2.0275795026940537,
 '댄디': 0.9740389515850403,
 '스포츠': 1.3001727949420643,
 '캐주얼': 0.43924459970068025,
 '포멀': 1.034823399672891,
 '로맨틱': 1.7414980203298451,
 '시크': 2.10806886564001,
 '스트릿': 1.1909603661268406,
 '걸리시': 1.4472806268213405}

In [48]:
from tqdm import tqdm

coordi_representation = pd.DataFrame(columns=sorted(total_coordi), index=musinsa_21.index)
for index, each_row in tqdm(musinsa_21.iterrows()):
    try:
        dict_temp = {i: coordi_count[i] for i in ast.literal_eval(each_row["coordi"])}
    except:
        dict_temp = {i: np.NaN}
    row_to_add = pd.DataFrame(dict_temp, index=[index])
    coordi_representation.update(row_to_add)

coordi_representation

28344it [00:56, 501.48it/s]


걸리시   골프   댄디  레트로  로맨틱      스트릿       스포츠   시크 아메리칸 캐주얼       캐주얼  \
0      NaN  NaN  NaN  NaN  NaN  1.19096  1.300173  NaN      NaN  0.439245   
1      NaN  NaN  NaN  NaN  NaN      NaN       NaN  NaN      NaN  0.439245   
2      NaN  NaN  NaN  NaN  NaN      NaN       NaN  NaN      NaN  0.439245   
3      NaN  NaN  NaN  NaN  NaN      NaN       NaN  NaN      NaN  0.439245   
4      NaN  NaN  NaN  NaN  NaN      NaN       NaN  NaN      NaN  0.439245   
...    ...  ...  ...  ...  ...      ...       ...  ...      ...       ...   
28339  NaN  NaN  NaN  NaN  NaN      NaN       NaN  NaN      NaN  0.439245   
28340  NaN  NaN  NaN  NaN  NaN      NaN       NaN  NaN      NaN  0.439245   
28341  NaN  NaN  NaN  NaN  NaN      NaN       NaN  NaN      NaN  0.439245   
28342  NaN  NaN  NaN  NaN  NaN      NaN       NaN  NaN      NaN       NaN   
28343  NaN  NaN  NaN  NaN  NaN  1.19096       NaN  NaN      NaN       NaN   

        포멀  홈웨어  
0      NaN  NaN  
1      NaN  NaN  
2      NaN  NaN  
3      NaN  NaN  
4      NaN  NaN  
...    ...  ...  
28339  NaN  NaN  
28340  NaN  NaN  
28341  NaN  NaN  
28342  NaN  NaN  
28343  NaN  NaN  

[28344 rows x 12 columns]

In [49]:
coordi_representation.fillna(0, inplace=True)
coordi_representation

걸리시   골프   댄디  레트로  로맨틱      스트릿       스포츠   시크  아메리칸 캐주얼       캐주얼  \
0      0.0  0.0  0.0  0.0  0.0  1.19096  1.300173  0.0       0.0  0.439245   
1      0.0  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245   
2      0.0  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245   
3      0.0  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245   
4      0.0  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245   
...    ...  ...  ...  ...  ...      ...       ...  ...       ...       ...   
28339  0.0  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245   
28340  0.0  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245   
28341  0.0  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245   
28342  0.0  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.000000   
28343  0.0  0.0  0.0  0.0  0.0  1.19096  0.000000  0.0       0.0  0.000000   

        포멀  홈웨어  
0      0.0  0.0  
1      0.0  0.0  
2      0.0  0.0  
3      0.0  0.0  
4      0.0  0.0  
...    ...  ...  
28339  0.0  0.0  
28340  0.0  0.0  
28341  0.0  0.0  
28342  0.0  0.0  
28343  0.0  0.0  

[28344 rows x 12 columns]

In [50]:
#유사도 평가 - cosine similarity

from sklearn.metrics.pairwise import cosine_similarity

def cos_sim_matrix(a, b):
    cos_sim = cosine_similarity(a, b)
    result_df = pd.DataFrame(data=cos_sim, index=[a.index])

    return result_df

In [51]:
print(coordi_representation.head())

   걸리시   골프   댄디  레트로  로맨틱      스트릿       스포츠   시크  아메리칸 캐주얼       캐주얼   포멀  \
0  0.0  0.0  0.0  0.0  0.0  1.19096  1.300173  0.0       0.0  0.439245  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245  0.0   
2  0.0  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245  0.0   
3  0.0  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245  0.0   
4  0.0  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245  0.0   

   홈웨어  
0  0.0  
1  0.0  
2  0.0  
3  0.0  
4  0.0  


In [52]:
cs_df = cos_sim_matrix(coordi_representation, coordi_representation)
cs_df.head()

0         1         2         3         4         5         6      \
0  1.000000  0.241731  0.241731  0.241731  0.241731  0.241731  0.241731   
1  0.241731  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000   
2  0.241731  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000   
3  0.241731  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000   
4  0.241731  1.000000  1.000000  1.000000  1.000000  1.000000  1.000000   

      7         8         9      ...     28334  28335     28336  28337  \
0  0.241731  0.241731  0.755259  ...  0.241731    0.0  0.241731    0.0   
1  1.000000  1.000000  0.320064  ...  1.000000    0.0  1.000000    0.0   
2  1.000000  1.000000  0.320064  ...  1.000000    0.0  1.000000    0.0   
3  1.000000  1.000000  0.320064  ...  1.000000    0.0  1.000000    0.0   
4  1.000000  1.000000  0.320064  ...  1.000000    0.0  1.000000    0.0   

      28338     28339     28340     28341  28342     28343  
0  0.241731  0.241731  0.241731  0.241731    0.0  0.655426  
1  1.000000  1.000000  1.000000  1.000000    0.0  0.000000  
2  1.000000  1.000000  1.000000  1.000000    0.0  0.000000  
3  1.000000  1.000000  1.000000  1.000000    0.0  0.000000  
4  1.000000  1.000000  1.000000  1.000000    0.0  0.000000  

[5 rows x 28344 columns]

In [53]:
print(cs_df[1].sort_values(ascending=False))

12665    1.0
25307    1.0
5705     1.0
25333    1.0
12211    1.0
        ... 
12763    0.0
12760    0.0
12759    0.0
12756    0.0
28343    0.0
Name: 1, Length: 28344, dtype: float64


코디 하나만으로 할 경우 1인 값이 많이 나오게 된다. 이것으로 두 품목이 유사하다고 볼수는 없다. 추가적으로 더 해봐야할 것같다. 

##### tags 해보자,,

In [54]:
#tags
import ast

a = []
num_na = 0

for i in musinsa_21["tags"]:
    li = ast.literal_eval(i)
    for j in li:
        a.append(j)
    if len(li) == 0:
        num_na += 1

In [55]:
tag_list = list(set(a))
tag_list

['박스',
 '비너스아시안핏스니커즈',
 '벨팬츠',
 '폴로로고모자',
 '울비니',
 '르',
 '테이퍼드핏',
 '남자가죽자켓',
 '필통',
 '야세×한국타이어',
 'COLDWEATHERCOLLECTION',
 '반스패치워크에라',
 '오프숄더긴팔',
 '거실화',
 '블라우스',
 '왁스가방',
 '가죽케어',
 'K121후드',
 'SL1AL71GA',
 '더블재킷',
 '베이직티',
 '퍼포먼스팬츠',
 '맨투맨추천',
 '라이풀컬래버레이션',
 '너란스니커즈',
 '키링카드지갑',
 'KANCO베레모',
 '모크화',
 '이지팬츠',
 '레이온셔츠',
 '420721CY1A',
 '프린트티셔츠',
 '나이키집업',
 '아이패드케이스',
 '커트코베인',
 '2911D2WY25',
 '무지반팔',
 '송치',
 '청키어글리슈즈',
 '미니멀가디건',
 '슬라이드',
 '아미트레이닝바지',
 '나이키후드자켓',
 '포켓긴팔티',
 '핸드백스트랩',
 '크로스백팩',
 '여자플랫슈즈',
 '숏블루종',
 '남성부츠',
 '컬러블록캡',
 '캐쥬얼샌들',
 '볼캡추천',
 '선물',
 'DANKERTECH',
 '키높이샌들',
 '로고가방',
 '비니',
 '츄리닝반바지',
 '상하의세트',
 '나이키에어맥스',
 '클래식숏팬츠',
 '레더토트백',
 '블랙청바지',
 '츄바스코말라가',
 '코닥패딩',
 '오버사이즈후드집업',
 'MINIMAL',
 '폴로럭비',
 '여자반팔티',
 '캔버스스니커즈',
 '컬러팬츠',
 '크루넥티셔츠',
 '슈퍼스타82',
 'VSW',
 'U넥니트',
 '카바스모노그램',
 '조거운동화',
 '포발란스컬렉션',
 '무지레깅스',
 '아모멘토원피스',
 '컨텐드',
 '커팅진',
 '트레포일컨버스',
 '그라미치팬츠',
 '21SUMMER컬렉션',
 '면팬츠',
 '22SPRING',
 '엄브로가방',
 '스트레이트데님',
 '르노이드페니로퍼',
 '누보선데이',
 '데일리에코백',
 '예쁜어글리',
 '바이커진'

In [56]:
import re

new_tag = []
for i in tag_list:
    if re.sub("[가-힣]","", i) == "":
        new_tag.append(i)

new_tag

['박스',
 '비너스아시안핏스니커즈',
 '벨팬츠',
 '폴로로고모자',
 '울비니',
 '르',
 '테이퍼드핏',
 '남자가죽자켓',
 '필통',
 '반스패치워크에라',
 '오프숄더긴팔',
 '거실화',
 '블라우스',
 '왁스가방',
 '가죽케어',
 '더블재킷',
 '베이직티',
 '퍼포먼스팬츠',
 '맨투맨추천',
 '라이풀컬래버레이션',
 '너란스니커즈',
 '키링카드지갑',
 '모크화',
 '이지팬츠',
 '레이온셔츠',
 '프린트티셔츠',
 '나이키집업',
 '아이패드케이스',
 '커트코베인',
 '무지반팔',
 '송치',
 '청키어글리슈즈',
 '미니멀가디건',
 '슬라이드',
 '아미트레이닝바지',
 '나이키후드자켓',
 '포켓긴팔티',
 '핸드백스트랩',
 '크로스백팩',
 '여자플랫슈즈',
 '숏블루종',
 '남성부츠',
 '컬러블록캡',
 '캐쥬얼샌들',
 '볼캡추천',
 '선물',
 '키높이샌들',
 '로고가방',
 '비니',
 '츄리닝반바지',
 '상하의세트',
 '나이키에어맥스',
 '클래식숏팬츠',
 '레더토트백',
 '블랙청바지',
 '츄바스코말라가',
 '코닥패딩',
 '오버사이즈후드집업',
 '폴로럭비',
 '여자반팔티',
 '캔버스스니커즈',
 '컬러팬츠',
 '크루넥티셔츠',
 '카바스모노그램',
 '조거운동화',
 '포발란스컬렉션',
 '무지레깅스',
 '아모멘토원피스',
 '컨텐드',
 '커팅진',
 '트레포일컨버스',
 '그라미치팬츠',
 '면팬츠',
 '엄브로가방',
 '스트레이트데님',
 '르노이드페니로퍼',
 '누보선데이',
 '데일리에코백',
 '예쁜어글리',
 '바이커진',
 '카고조거',
 '여성스커트',
 '칼하트파우치',
 '오가닉코튼',
 '레더백컬렉션',
 '슬리퍼',
 '폴로월렛',
 '웨딩구두',
 '노트북파우치',
 '스트레이트핏데님팬츠',
 '세미오버핏',
 '반지갑',
 '남녀공용티',
 '바람막이점퍼',
 '휴대폰케이스',
 '가을점퍼',
 '포켓자켓',
 '언폴드가방',
 '여자운동복'

In [57]:
tag_count = dict.fromkeys(new_tag)

for each_tag in musinsa_21['tags']:
    for tag in ast.literal_eval(each_tag):
        try:
            if tag_count[tag] == None:
                tag_count[tag] = 1
            else:
                tag_count[tag] = tag_count[tag]+1
        except:
            pass

In [82]:
tag_count

{'박스': 2,
 '비너스아시안핏스니커즈': 1,
 '벨팬츠': 1,
 '폴로로고모자': 10,
 '울비니': 13,
 '르': 1,
 '테이퍼드핏': 93,
 '남자가죽자켓': 12,
 '필통': 1,
 '반스패치워크에라': 1,
 '오프숄더긴팔': 1,
 '거실화': 10,
 '블라우스': 26,
 '왁스가방': 1,
 '가죽케어': 1,
 '더블재킷': 3,
 '베이직티': 49,
 '퍼포먼스팬츠': 1,
 '맨투맨추천': 31,
 '라이풀컬래버레이션': 4,
 '너란스니커즈': 28,
 '키링카드지갑': 1,
 '모크화': 9,
 '이지팬츠': 67,
 '레이온셔츠': 2,
 '프린트티셔츠': 1,
 '나이키집업': 3,
 '아이패드케이스': 41,
 '커트코베인': 1,
 '무지반팔': 15,
 '송치': 2,
 '청키어글리슈즈': 3,
 '미니멀가디건': 2,
 '슬라이드': 286,
 '아미트레이닝바지': 2,
 '나이키후드자켓': 1,
 '포켓긴팔티': 1,
 '핸드백스트랩': 1,
 '크로스백팩': 2,
 '여자플랫슈즈': 1,
 '숏블루종': 1,
 '남성부츠': 26,
 '컬러블록캡': 2,
 '캐쥬얼샌들': 8,
 '볼캡추천': 113,
 '선물': 27,
 '키높이샌들': 2,
 '로고가방': 2,
 '비니': 264,
 '츄리닝반바지': 3,
 '상하의세트': 1,
 '나이키에어맥스': 2,
 '클래식숏팬츠': 1,
 '레더토트백': 1,
 '블랙청바지': 1,
 '츄바스코말라가': 2,
 '코닥패딩': 9,
 '오버사이즈후드집업': 8,
 '폴로럭비': 6,
 '여자반팔티': 59,
 '캔버스스니커즈': 8,
 '컬러팬츠': 1,
 '크루넥티셔츠': 1,
 '카바스모노그램': 7,
 '조거운동화': 2,
 '포발란스컬렉션': 2,
 '무지레깅스': 5,
 '아모멘토원피스': 1,
 '컨텐드': 3,
 '커팅진': 2,
 '트레포일컨버스': 2,
 '그라미치팬츠': 14,
 '면팬츠': 144,
 '엄브로가방': 2,
 '스트레이트데

In [58]:
new_tag_count = {}

for i in tag_count:
    if tag_count[i] >= 10:
        temp =  new_tag_count.get(i)
        if temp == None:
            new_tag_count[i] = tag_count[i]

In [59]:
len(new_tag_count.keys())

1619

In [61]:
for each_tag in new_tag_count:
    new_tag_count[each_tag] = np.log10(total_count/new_tag_count[each_tag]) 
  
new_tag_count 

{'폴로로고모자': 3.914332340436985,
 '울비니': 3.928907444907091,
 '테이퍼드핏': 4.057313360891959,
 '남자가죽자켓': 3.92440876165124,
 '거실화': 3.914332340436985,
 '블라우스': 3.969946598285209,
 '베이직티': 4.011195789388548,
 '맨투맨추천': 3.981008146355715,
 '너란스니커즈': 3.9745728588313503,
 '이지팬츠': 4.03310252894899,
 '아이패드케이스': 3.9991921522172986,
 '무지반팔': 3.937068214523608,
 '슬라이드': 4.152279909431812,
 '남성부츠': 3.969946598285209,
 '볼캡추천': 4.072361616894128,
 '선물': 3.972296415767899,
 '비니': 4.144781749658717,
 '여자반팔티': 4.02406566411733,
 '그라미치팬츠': 3.93311453395291,
 '면팬츠': 4.09184905050858,
 '카고조거': 3.959695379812004,
 '여성스커트': 4.0393058937215125,
 '슬리퍼': 4.182101583009219,
 '노트북파우치': 4.022864983835588,
 '세미오버핏': 3.919570704301002,
 '반지갑': 4.053137335885438,
 '바람막이점퍼': 4.0362596244412,
 '드레스': 3.9745728588313503,
 '프라임그린': 3.92440876165124,
 '대두모자': 3.9789245552849954,
 '포인트백': 3.990638792448027,
 '르삭라인': 3.937068214523608,
 '고프코어': 3.969946598285209,
 '구스다운': 4.070251598443764,
 '겨울': 3.985008845494423,
 '블랙로퍼': 3.940

In [62]:
total_tag = list(new_tag_count.keys())

In [63]:
tag_representation = pd.DataFrame(columns=sorted(total_tag), index=musinsa_21.index)
for index, each_row in tqdm(musinsa_21.iterrows()):
    try:
        dict_temp = {i: new_tag_count[i] for i in ast.literal_eval(each_row['tags'])}
    except:
        pass
    row_to_add = pd.DataFrame(dict_temp, index=[index])
    tag_representation.update(row_to_add)

tag_representation

28344it [5:40:11,  1.39it/s] 


가디건        가방 가방스트랩 가방추천 가방컬렉션   가을 가을가디건 가을니트 가을바지 가을베스트  ... 후드티추천  \
0      NaN       NaN   NaN  NaN   NaN  NaN   NaN  NaN  NaN   NaN  ...   NaN   
1      NaN       NaN   NaN  NaN   NaN  NaN   NaN  NaN  NaN   NaN  ...   NaN   
2      NaN       NaN   NaN  NaN   NaN  NaN   NaN  NaN  NaN   NaN  ...   NaN   
3      NaN       NaN   NaN  NaN   NaN  NaN   NaN  NaN  NaN   NaN  ...   NaN   
4      NaN  4.259079   NaN  NaN   NaN  NaN   NaN  NaN  NaN   NaN  ...   NaN   
...    ...       ...   ...  ...   ...  ...   ...  ...  ...   ...  ...   ...   
28339  NaN       NaN   NaN  NaN   NaN  NaN   NaN  NaN  NaN   NaN  ...   NaN   
28340  NaN       NaN   NaN  NaN   NaN  NaN   NaN  NaN  NaN   NaN  ...   NaN   
28341  NaN       NaN   NaN  NaN   NaN  NaN   NaN  NaN  NaN   NaN  ...   NaN   
28342  NaN       NaN   NaN  NaN   NaN  NaN   NaN  NaN  NaN   NaN  ...   NaN   
28343  NaN       NaN   NaN  NaN   NaN  NaN   NaN  NaN  NaN   NaN  ...   NaN   

        후디 후디집업  후리스 후리스자켓 후리스점퍼 후리스집업   흑청   힙색   힙합  
0      NaN  NaN  NaN   NaN   NaN   NaN  NaN  NaN  NaN  
1      NaN  NaN  NaN   NaN   NaN   NaN  NaN  NaN  NaN  
2      NaN  NaN  NaN   NaN   NaN   NaN  NaN  NaN  NaN  
3      NaN  NaN  NaN   NaN   NaN   NaN  NaN  NaN  NaN  
4      NaN  NaN  NaN   NaN   NaN   NaN  NaN  NaN  NaN  
...    ...  ...  ...   ...   ...   ...  ...  ...  ...  
28339  NaN  NaN  NaN   NaN   NaN   NaN  NaN  NaN  NaN  
28340  NaN  NaN  NaN   NaN   NaN   NaN  NaN  NaN  NaN  
28341  NaN  NaN  NaN   NaN   NaN   NaN  NaN  NaN  NaN  
28342  NaN  NaN  NaN   NaN   NaN   NaN  NaN  NaN  NaN  
28343  NaN  NaN  NaN   NaN   NaN   NaN  NaN  NaN  NaN  

[28344 rows x 1619 columns]

In [67]:
tag_representation.shape

(28344, 1619)

- tag와 coordi 합쳐보기 

In [70]:
musinsa_representation = pd.concat([tag_representation, coordi_representation], axis=1).fillna(0)
print(musinsa.shape)

(57491, 25)


In [71]:
musinsa_representation

가디건        가방  가방스트랩  가방추천  가방컬렉션   가을  가을가디건  가을니트  가을바지  가을베스트  ...  \
0      0.0  0.000000    0.0   0.0    0.0  0.0    0.0   0.0   0.0    0.0  ...   
1      0.0  0.000000    0.0   0.0    0.0  0.0    0.0   0.0   0.0    0.0  ...   
2      0.0  0.000000    0.0   0.0    0.0  0.0    0.0   0.0   0.0    0.0  ...   
3      0.0  0.000000    0.0   0.0    0.0  0.0    0.0   0.0   0.0    0.0  ...   
4      0.0  4.259079    0.0   0.0    0.0  0.0    0.0   0.0   0.0    0.0  ...   
...    ...       ...    ...   ...    ...  ...    ...   ...   ...    ...  ...   
28339  0.0  0.000000    0.0   0.0    0.0  0.0    0.0   0.0   0.0    0.0  ...   
28340  0.0  0.000000    0.0   0.0    0.0  0.0    0.0   0.0   0.0    0.0  ...   
28341  0.0  0.000000    0.0   0.0    0.0  0.0    0.0   0.0   0.0    0.0  ...   
28342  0.0  0.000000    0.0   0.0    0.0  0.0    0.0   0.0   0.0    0.0  ...   
28343  0.0  0.000000    0.0   0.0    0.0  0.0    0.0   0.0   0.0    0.0  ...   

        댄디  레트로  로맨틱      스트릿       스포츠   시크  아메리칸 캐주얼       캐주얼   포멀  홈웨어  
0      0.0  0.0  0.0  1.19096  1.300173  0.0       0.0  0.439245  0.0  0.0  
1      0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245  0.0  0.0  
2      0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245  0.0  0.0  
3      0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245  0.0  0.0  
4      0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245  0.0  0.0  
...    ...  ...  ...      ...       ...  ...       ...       ...  ...  ...  
28339  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245  0.0  0.0  
28340  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245  0.0  0.0  
28341  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.439245  0.0  0.0  
28342  0.0  0.0  0.0  0.00000  0.000000  0.0       0.0  0.000000  0.0  0.0  
28343  0.0  0.0  0.0  1.19096  0.000000  0.0       0.0  0.000000  0.0  0.0  

[28344 rows x 1631 columns]

In [74]:
musinsa_cs_df = cos_sim_matrix(musinsa_representation, musinsa_representation)

In [73]:
musinsa_cs_df

0         1         2         3         4         5         6      \
0      1.000000  0.008106  0.008106  0.008106  0.008169  0.008813  0.008205   
1      0.008106  1.000000  1.000000  1.000000  0.149377  0.457494  0.150045   
2      0.008106  1.000000  1.000000  1.000000  0.149377  0.457494  0.150045   
3      0.008106  1.000000  1.000000  1.000000  0.149377  0.457494  0.150045   
4      0.008169  0.149377  0.149377  0.149377  1.000000  0.327991  0.442751   
...         ...       ...       ...       ...       ...       ...       ...   
28339  0.008090  0.001604  0.001604  0.001604  0.001617  0.001744  0.001624   
28340  0.008090  0.001604  0.001604  0.001604  0.001617  0.001744  0.001624   
28341  0.008090  0.001604  0.001604  0.001604  0.001617  0.001744  0.001624   
28342  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   
28343  0.059174  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   

          7         8         9      ...     28334  28335     28336     28337  \
0      0.008205  0.008205  0.079525  ...  0.010541    0.0  0.008090  0.000000   
1      0.150045  0.150045  0.148973  ...  0.002090    0.0  0.001604  0.000000   
2      0.150045  0.150045  0.148973  ...  0.002090    0.0  0.001604  0.000000   
3      0.150045  0.150045  0.148973  ...  0.002090    0.0  0.001604  0.000000   
4      0.442751  0.442751  0.439590  ...  0.002107    0.0  0.001617  0.000000   
...         ...       ...       ...  ...       ...    ...       ...       ...   
28339  0.001624  0.001624  0.001612  ...  0.002086    0.0  1.000000  0.999199   
28340  0.001624  0.001624  0.001612  ...  0.002086    0.0  1.000000  0.999199   
28341  0.001624  0.001624  0.001612  ...  0.002086    0.0  1.000000  0.999199   
28342  0.000000  0.000000  0.000000  ...  0.000000    0.0  0.999199  1.000000   
28343  0.000000  0.000000  0.000000  ...  0.000000    0.0  0.993360  0.994157   

          28338     28339     28340     28341     28342     28343  
0      0.008090  0.008090  0.008090  0.008090  0.000000  0.059174  
1      0.001604  0.001604  0.001604  0.001604  0.000000  0.000000  
2      0.001604  0.001604  0.001604  0.001604  0.000000  0.000000  
3      0.001604  0.001604  0.001604  0.001604  0.000000  0.000000  
4      0.001617  0.001617  0.001617  0.001617  0.000000  0.000000  
...         ...       ...       ...       ...       ...       ...  
28339  1.000000  1.000000  1.000000  1.000000  0.999199  0.993360  
28340  1.000000  1.000000  1.000000  1.000000  0.999199  0.993360  
28341  1.000000  1.000000  1.000000  1.000000  0.999199  0.993360  
28342  0.999199  0.999199  0.999199  0.999199  1.000000  0.994157  
28343  0.993360  0.993360  0.993360  0.993360  0.994157  1.000000  

[28344 rows x 28344 columns]

In [75]:
musinsa_cs_df[1].sort_values(ascending=False)

2        1.000000
3        1.000000
1        1.000000
75       0.929585
76       0.923294
           ...   
13609    0.000000
13608    0.000000
13607    0.000000
13606    0.000000
28343    0.000000
Name: 1, Length: 28344, dtype: float64

In [81]:
print(musinsa_21.loc[2, ["brand", "main_category","sub_category","coordi","tags"]])
print(musinsa_21.loc[3, ["brand", "main_category","sub_category","coordi","tags"]])
print(musinsa_21.loc[1, ["brand", "main_category","sub_category","coordi","tags"]])
print(musinsa_21.loc[75, ["brand", "main_category","sub_category","coordi","tags"]])
print(musinsa_21.loc[76, ["brand", "main_category","sub_category","coordi","tags"]])

brand                                                    NEIKIDNIS
main_category                                                   가방
sub_category                                                    백팩
coordi                                                     ['캐주얼']
tags             ['메쉬스트링백팩', '스트링백팩', 'MESHSTRINGBACKPACK', '코듀...
Name: 2, dtype: object
brand                                                    NEIKIDNIS
main_category                                                   가방
sub_category                                                    백팩
coordi                                                     ['캐주얼']
tags             ['백팩', '앱솔루트', '학생가방', '코듀라', '포켓', '노트북', '커플...
Name: 3, dtype: object
brand                                                         YALE
main_category                                                   가방
sub_category                                                    백팩
coordi                                                     ['캐주얼']
tags            

In [124]:
musinsa_cs_df[3000].sort_values(ascending=False)

2994     1.0
2995     1.0
2996     1.0
3000     1.0
2997     1.0
        ... 
9752     0.0
9751     0.0
9750     0.0
9749     0.0
28343    0.0
Name: 3000, Length: 28344, dtype: float64

In [125]:
print(musinsa_21.loc[3000, ["brand", "main_category","sub_category","coordi","tags"]])
print(musinsa_21.loc[2994, ["brand", "main_category","sub_category","coordi","tags"]])
print(musinsa_21.loc[2995, ["brand", "main_category","sub_category","coordi","tags"]])
print(musinsa_21.loc[2996, ["brand", "main_category","sub_category","coordi","tags"]])
print(musinsa_21.loc[2997, ["brand", "main_category","sub_category","coordi","tags"]])

brand                                                      TANTIN
main_category                                                  가방
sub_category                                               브리프케이스
coordi                                                     ['포멀']
tags             ['서류가방', '남자서류가방', '브리프케이스', '남성서류가방', '가죽서류가방']
Name: 3000, dtype: object
brand                                                      TANTIN
main_category                                                  가방
sub_category                                               브리프케이스
coordi                                                     ['포멀']
tags             ['서류가방', '브리프케이스', '남자서류가방', '남성서류가방', '가죽서류가방']
Name: 2994, dtype: object
brand                                                    LOTUFF
main_category                                                가방
sub_category                                             브리프케이스
coordi                                                   ['포멀']
tags             ['가죽', '가죽크로스백'